### PR TITLE
feat: Adding new /datafile endpoint

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Providing datafile access through `/datafile` endpoint

## Issues
- https://optimizely.atlassian.net/browse/OASIS-5500

### CODE REVIEW

File: api/openapi-spec/openapi.yaml [hunk 1]
```diff
@@ -28,6 +28,23 @@ paths:
...
+      operationId: getDatafile
+      description: Returns the json datafile for the given environment.
+      responses:
+        '200':
+          description: Valid response
+          content:
+            application/json:
+              schema:
+                description: Optimizely Datafile
+                type: object
```
Comment (msohailhussain) [inline api/openapi-spec/openapi.yaml:43]: should it be object or string?

**Review by msohailhussain — APPROVED** [review]
have one comment, otherwise looks good.

**Comment by msohailhussain** [comment]
Before merging, need to check with FSC.

File: api/openapi-spec/openapi.yaml (hunk 1 above)
Comment (yasirfolio3) [reply api/openapi-spec/openapi.yaml:43]: We are returning datafile as json hence have written type as `object` here. If we intend to return datafile as `string` then we will have to map it against a key map to render it as json in response.

File: pkg/handlers/get_datafile_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+package handlers
```
Comment (mjc1283) [inline pkg/handlers/get_datafile_test.go:1]: Add license header

**Review by mjc1283 — APPROVED** [review]
LGTM. Please add license header to get_datafile_test.go.

**Pull request merged**